src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/configuration/KafkaConfiguration.java

```diff
@@ -71,32 +75,23 @@ ConsumerConfig getMainConsumerConfig() {
 
     @Bean("retry-consumer-group")
     @Lazy
-    CHKafkaConsumerGroup getRetryConsumer() {
-        return new CHKafkaConsumerGroup(getRetryConsumerConfig());
+    CHKafkaResilientConsumerGroup getRetryConsumer() {
+        return new CHKafkaResilientConsumerGroup(getRetryConsumerConfig(), CHConsumerType.RETRY_CONSUMER);
     }
 
     ConsumerConfig getRetryConsumerConfig() {
-        ConsumerConfig config = new ConsumerConfig();
+        ConsumerConfig config = new ConsumerConfig(APPLICATION_NAME);
         config.setBrokerAddresses(new String[]{brokerAddress});
-        config.setTopics(Collections.singletonList(retryTopicName));
+        config.setTopics(Collections.singletonList(mainTopicName));
         config.setPollTimeout(pollTimeout);
         config.setGroupName(retryConsumerGroupName);
         return config;
     }
 
     @Bean("error-consumer-group")
     @Lazy
-    CHKafkaConsumerGroup getErrorConsumer() {
-        return new CHKafkaConsumerGroup(getErrorConsumerConfig());
-    }
-
-    ConsumerConfig getErrorConsumerConfig() {
-        ConsumerConfig config = new ConsumerConfig();
-        config.setBrokerAddresses(new String[]{brokerAddress});
-        config.setTopics(Collections.singletonList(errorTopicName));
-        config.setPollTimeout(pollTimeout);
-        config.setGroupName(errorConsumerGroupName);
-        return config;
+    CHKafkaResilientConsumerGroup getErrorConsumer() {
+        return new CHKafkaResilientConsumerGroup(getMainConsumerConfig(), CHConsumerType.ERROR_CONSUMER);
```

Comment:
what happens with the group name here? does the error consumer get added to the main consumer group?